### DISCUSSION TITLE
Why might a function work in one shop but not in another?

### DISCUSSION THREAD

**Discussion by sergiop14** [Shopify/function-examples#252] [Feedback: Discounts APIs]
We have developed custom discount functions and tested them on many shops but found that for some Shopify PLUS shops, the functions do not work.

We have ruled out that this is due to theme incompatibilities as we have tried other Shopify default themes on the shop itself and it still doesn't work.

We have also disabled all scripts and other apps in case any of these might conflict.

Even so, we were unable to find the source of the problem. Nor have we seen anything in the documentation that indicates possible incompatibilities apart from checkout.liquid.

As for the type of discount, we have tested all the discounts we have, both automatic and code-based, but none of the discounts from Shopify Functions work.

On the other hand, the logs of the functions do not show the executions of shops other than our development ones. Is this normal?
![Screenshot 2023-06-01 at 19 14 31](https://github.com/Shopify/function-examples/assets/66000223/289c2280-0ab9-4d8f-aed6-439592387555)

**Comment by DuncanUszkay1** [comment]
👋🏻 Hey Sergio. We only make logs from production shops available to function developers when the function fails (times out, writes to STDERR, etc). In those cases, the merchant can submit that error report to the developer and the logs will appear in this view.

It's certainly possible for a function to work on one shop but not another, as functions can load shop-bound configuration, most notably through metafields, but also through the contents of the cart. My recommendation would be to force the function to raise an error and have the merchant send you the error report as explained [here](https://shopify.dev/docs/api/functions/errors#error-reports). Then you'll be able to use the logs to determine what in the input has caused the discount to not apply.

**Reply by sergiop14** [reply]
Hey Duncan! Thanks for your help!

We have been testing all day what you have told us because the truth is that we didn't know that it works like this.

We have included logs and we have made the function fail whenever it executes, but to our surprise, it gives an error, so we understand that it does not even execute.

The discount is created in the customer's shop but without errors:

<img width="1468" alt="image" src="https://github.com/Shopify/function-examples/assets/66000223/1d887cbf-7b63-4e77-8c66-97f767158707">

How is this possible, how can we check what the problem is?

**Reply by DuncanUszkay1** [reply]
That is odd 🤔 it's difficult for me to look into this further without more information. I see you've omitted the IDs from your screenshots for privacy reasons. 

Some other things you could do on your end:
- Ensure that the function you updated with the error is indeed the one on the shop by comparing the ID in your `.env` file with the function_id on the [discount app type](https://shopify.dev/docs/api/admin-graphql/unstable/objects/AppDiscountType#field-appdiscounttype-functionid) of that automatic discount you created

**Reply by sergiop14** [reply]
We have checked and it matches. We don't know what else to test. Are there any settings that only PLUS stores have that might conflict? e.g. Markets, POS, or something like that?

It has to be something at the general level of the store not just the cart, because if it were only the cart would work in the checkout to create the function as a coupon, and also does not run.

![image](https://github.com/Shopify/function-examples/assets/66000223/9e94b0d6-c055-440d-929d-e60ee000b122)

**Reply by DuncanUszkay1** [reply]
Do non-app discounts apply properly? 🤔

**Comment by sergiop14** [comment]
Yes, they work fine, and if we create from the API any Shopify native discount, it works fine.

Maybe there is a conflict application?

(Planet is our app) There are none that use functions except ours.

All scrips are in draft. All flows are about 'order created’… I don’t know Dunk, this is very strange

￼

**Comment by sergiop14** [comment]
How can I access it? with my Shopify dev account returns me an error

￼

**Reply by DuncanUszkay1** [reply]
Ah sorry I thought that was a public link. In short, it states that the shop might be excluded from extensible discounts. Without knowing what shop we're talking about, I can't go any further.

**Reply by DuncanUszkay1** [reply]
You might have more luck reaching out to partner support: https://community.shopify.com/c/shopify-community/ct-p/en

**Comment by sergiop14** [comment]
Right now we manage 3 PLUS stores that have the same problem with the functions:

https://canwest-esthetics.myshopify.com/￼
Creata Beauty
canwest-esthetics.myshopify.com

https://lcitest1.myshopify.com/￼

http://labelleboucle-2022.myshopify.com￼
Produits Naturels pour Cheveux Bouclés
labelleboucle-2022.myshopify.com


Is there any way to activate the extensible discounts on that stores?

Thanks Dunk!

**Comment by sergiop14** [comment]
Dunk, it was that!! I’ve been talking with the support; they enabled it for my managed stores, and now it’s working.

Thank you very much for your help, you were the only one who wanted to help me!!

**Reply by DuncanUszkay1** [reply]
I'm glad it worked out! Best of luck using functions 😄
